Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1005,6 +1050,76 @@ getStats (driverId, merchantId) date = do
            in Money $ GHCL.sum driverSelFares' + GHCL.sum customerExtFees' + GHCL.sum deadKmFare'
       }
 
+driverPhotoUpload :: (Id SP.Person, Id DM.Merchant) -> DriverPhotoUploadReq -> Flow APISuccess
+driverPhotoUpload (driverId, merchantId) DriverPhotoUploadReq {..} = do
+  person <- runInReplica $ QPerson.findById driverId >>= fromMaybeM (PersonNotFound (getId driverId))
+  encImage <- L.runIO $ base64Encode <$> BS.readFile image
+  imageExtension <- validateContentType
+  let req = IF.ImageVerificationReq {file = encImage}
+  _ <- validateFaceImage merchantId req
+  filePath <- createFilePath (getId driverId) fileType imageExtension
+  transporterConfig <- CQTC.findByMerchantId (person.merchantId) >>= fromMaybeM (TransporterConfigNotFound (getId (person.merchantId)))
+  let fileUrl =
+        transporterConfig.mediaFileUrlPattern
+          & T.replace "<DOMAIN>" "driver-profile-picture"
+          & T.replace "<FILE_PATH>" filePath
+  result <- try @_ @SomeException $ S3.put (T.unpack filePath) encImage
+  case result of
+    Left err -> throwError $ InternalError ("S3 Upload Failed: " <> show err)
+    Right _ -> do
+      case person.mediaId of
+        Just mediaFileId -> do
+          Esq.runTransaction $ do
+            QPerson.updateMediaId driverId Nothing
+            MFQuery.deleteById mediaFileId
+        Nothing -> return ()
+      createMediaEntry driverId Common.AddLinkAsMedia {url = fileUrl, fileType}
+  where
+    validateContentType = do
+      case fileType of
+        Common.Image -> case reqContentType of
+          "image/png" -> pure "png"
+          "image/jpeg" -> pure "jpg"
+          _ -> throwError $ FileFormatNotSupported reqContentType
+        _ -> throwError $ FileFormatNotSupported reqContentType
+
+createFilePath ::
+  (MonadTime m, MonadReader r m, HasField "s3Env" r (S3.S3Env m)) =>
+  Text ->
+  Common.FileType ->
+  Text ->
+  m Text
+createFilePath driverId fileType imageExtension = do
```

Comment:
All these function are already there, why don't you use them. Use `DriverOnboading/Image` validate api

Reply:
Created a new function as there are different functions for different file paths. Should I keep the function or remove it?